test/Dapr.E2E.Test/Workflows/WorkflowTest.cs

```diff
@@ -41,19 +42,69 @@ public async Task TestWorkflows()
+                workflowComponent: workflowComponent,
+                workflowName: workflowName,
+                input: input,
+                workflowOptions: workflowOptions,
+                cancellationToken: cts);
+
+            startResponse.InstanceId.Should().Be("testInstanceId", $"Instance ID {startResponse.InstanceId} was not correct");
+
 
             // GET INFO TEST
-            var getResponse = await daprClient.GetWorkflowAsync(instanceId, workflowComponent, workflowName);
-            getResponse.instanceId.Should().Be("TestWorkflowInstanceID");
-            getResponse.metadata["dapr.workflow.runtime_status"].Should().Be("RUNNING", $"Instance ID {getResponse.metadata["dapr.workflow.runtime_status"]} was not correct");
+            var getResponse = await daprClient.GetWorkflowAsync(instanceId, workflowComponent, cts);
+            getResponse.instanceId.Should().Be("testInstanceId");
+            getResponse.runtimeStatus.Should().Be("RUNNING", $"Instance ID {getResponse.runtimeStatus} was not correct");
+
+            // PAUSE TEST:
+            await daprClient.PauseWorkflowAsync(instanceId, workflowComponent, cts);
+            getResponse = await daprClient.GetWorkflowAsync(instanceId, workflowComponent, cts);
+            getResponse.runtimeStatus.Should().Be("SUSPENDED", $"Instance ID {getResponse.runtimeStatus} was not correct");
+
+            // RESUME TEST:
+            await daprClient.ResumeWorkflowAsync(instanceId, workflowComponent, cts);
+            getResponse = await daprClient.GetWorkflowAsync(instanceId, workflowComponent, cts);
+            getResponse.runtimeStatus.Should().Be("RUNNING", $"Instance ID {getResponse.runtimeStatus} was not correct");
+
+            // RAISE EVENT TEST
+            await daprClient.RaiseWorkflowEventAsync(instanceId, workflowComponent, "ChangePurchaseItem", "computers", cts);
+            getResponse = await daprClient.GetWorkflowAsync(instanceId, workflowComponent, cts);
 
             // TERMINATE TEST:
-            await daprClient.TerminateWorkflowAsync(instanceId, workflowComponent);
-            getResponse = await daprClient.GetWorkflowAsync(instanceId, workflowComponent, workflowName);
-            getResponse.metadata["dapr.workflow.runtime_status"].Should().Be("TERMINATED", $"Instance ID {getResponse.metadata["dapr.workflow.runtime_status"]} was not correct");
+            await daprClient.TerminateWorkflowAsync(instanceId, workflowComponent, cts);
+            getResponse = await daprClient.GetWorkflowAsync(instanceId, workflowComponent, cts);
+            getResponse.runtimeStatus.Should().Be("TERMINATED", $"Instance ID {getResponse.runtimeStatus} was not correct");
+
+            // PURGE TEST
+            await daprClient.PurgeWorkflowAsync(instanceId, workflowComponent, cts);
+
+            try 
+            {
+                getResponse = await daprClient.GetWorkflowAsync(instanceId, workflowComponent, cts);
+            }
```

Comment:
Do we need to fail the test here if `GetWorkflowAsync` returns successfully?

Reply:
I'm confused. Is the expectation that we return a response with an empty instance ID or do we expect a `DaprException`? I feel like only one of those outcomes should be expected but the test allows for either one.

Based on the previous iteration, I would expecting something along the lines of `Assert.True(false)` inside the `try` block. If instead the design is that we return a response with an empty instance ID, then I think we should remove the try/catch.